src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -51,47 +57,100 @@ def setup(self, builder: Builder):
+        )
+        target_standard_housing_pop_size = (
+            self.pop_config.population_size - target_gq_pop_size
+        )
 
-    def generate_initial_households(self, pop_data: SimulantData) -> pd.DataFrame():
-        households = self.population_view.subview(["household_id"]).get(pop_data.index)
-        households = (
-            households.drop_duplicates().sort_values("household_id").set_index("household_id")
+        sampled_census_ids = vectorized_choice(
+            options=input_household_data["census_household_id"],
+            n_to_choose=target_standard_housing_pop_size,
+            weights=input_household_data["household_weight"],
+            additional_key=f"sample_households_{self.seed}",
         )
-        households["address_id"] = np.arange(len(households))
-        households["housing_type"] = households.index.map(GQ_HOUSING_TYPE_MAP).fillna(
-            "Standard"
+        standard_household_ids = gq_households.index.size + np.arange(len(sampled_census_ids))
+        sampled_standard_households = pd.DataFrame(
+            {
+                "census_household_id": sampled_census_ids,
+                "household_id": standard_household_ids,
+                "address_id": standard_household_ids,
+                "housing_type": "Standard",
+            }
         )
-        # set housing type dtype
+
+        households = pd.concat([gq_households, sampled_standard_households])
         households["housing_type"] = households["housing_type"].astype(
-            pd.CategoricalDtype(categories=HOUSING_TYPES)
+            pd.CategoricalDtype(categories=data_values.HOUSING_TYPES)
         )
 
         return households
 
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        """
+        Remove unused households from the household datastructure immediately
+        following initialization and set household id as index
+        """
+        if pop_data.creation_time < self.start_time:
+            households = self._households.set_index("household_id")
+            # Drop all households that aren't in population except for GQ households
+            gq_index = households.index[households["housing_type"] != "Standard"]
+            existing_households = pd.Index(
+                self.population_view.subview(["household_id"])
+                .get(pop_data.index)["household_id"]
+                .drop_duplicates()
+            )
+            self.set_households(households.loc[gq_index.union(existing_households)])
+
+    ##################################
+    # Pipeline sources and modifiers #
+    ##################################
+
     def get_household_details(self, idx: pd.Index) -> pd.DataFrame:
         """Source of the household_details pipeline"""
-        pop = self.population_view.get(idx)[["household_id"]]
+        pop = self.population_view.subview(["tracked", "household_id"]).get(idx)
```

Comment:
Won't this add a `tracked` column within the pipeline?

Reply:
You're right. I should remove this